lib/connectors/memory.js

```diff
@@ -237,6 +244,38 @@ Memory.prototype.updateOrCreate = function (model, data, options, callback) {
   });
 };
 
+Memory.prototype.findOrCreate = function(model, filter, data, callback) {
+  var self = this;
+  var nodes = self._findAll(model, filter);
+
+  if (nodes.length > 0) {
+    if (filter && filter.include) {
+      self._models[model].model.include(nodes[0], filter.include, {}, function(err, nodes) {
+        process.nextTick(function() {
+          if (err)  callback(err);
+          else
+            callback(null, nodes[0], false);
```

Comment:
Style: too many spaces after `if (err)`, plus inconsistency in which statements are on the same line vs. on a new line.

We usually use one of these two ways:

```js
if (err) return callback(err);
callback(null, nodes[0], false);

// or

if (err)
  callback(err);
else
  callback(null, nodes[0], false);
```